docarray/doc_index/abstract_doc_index.py

```diff
@@ -342,14 +344,22 @@ def __getitem__(
             return_singleton = False
         # retrieve data
         doc_sequence = self._get_items(key)
+        # check data
+        if return_singleton and len(doc_sequence) == 0:
```

Comment:
Why is this check only relevant if `return_singleton` is true?

Reply:
Because the last line is `return out_da[0] if return_singleton else out_da`.